index.js

```diff
@@ -38,18 +97,46 @@
  * And returns the first and last name concatenated together (i.e. 'Bob Evans').
  */
 
+function getFullName(name) {
+	var name = {
+		firstName: 'Bob',
+		lastName: 'Evans'
+	};
+	console.log(name.firstName + ' ' + name.lastName);
+	return name.firstName + ' ' + name.lastName;
+};
+
+// this one is not working. 
 
 /**
  * Q6: Write a function called `reverseString` that has a String parameter and returns
  * the String reversed. Hint: look up `.split()`
  */
 
 
+function reverseString(str) {
+	var strReverse = str.split('').reverse().join('');
+	console.log(strReverse);
+	return strReverse;
+};
+
+reverseString('asdfghjkl');
+
 /**
  * Q7: Write a function called `maxArray` that has one Array of Numbers parameter and returns
  * the largest number in the Array. Use a `forEach` loop.
  */
 
+function maxArray(arr) {
+	arr.forEach(function (largest) {
+		return Math.max.apply(Math, arr)
+		console.log(Math.max.apply(Math, arr))
+	});
```

Comment:
I think we talked about this in class, but when using `forEach` we generally don't return anything from the function that we pass in. The reason for this is that the `forEach` function itself does not return anything, so that value will get swallowed into the void.

The way we get around this is by, instead of returning a value, modifying a value contained outside the `forEach` function's scope.

```
function maxArray(arr) {
   // This value is outside of the scope of the forEach function. 
   // It can be accessed by the forEach function and it exists when it is done.
   var largestSoFar = arr[0];

   // now loop, modifying largestSoFar as we go
   arr.forEach(function(num) {
       if (num > largestSoFar) {
           largestSoFar = num;
       }
   }

   return largestSoFar;
}
```